stakerservice/service.go

```diff
@@ -125,6 +128,77 @@ func (s *StakerService) stake(_ *rpctypes.Context,
 	}, nil
 }
 
+func (s *StakerService) btcDelegationFromBtcStakingTx(
+	_ *rpctypes.Context,
+	stakerAddress string,
+	btcStkTxHash string,
+	globalParams *parser.GlobalParams,
+) (*ResultBtcDelegationFromBtcStakingTx, error) {
+	stkTxHash, err := chainhash.NewHashFromStr(btcStkTxHash)
+	if err != nil {
+		s.logger.WithError(err).Info("err parse tx hash")
+		return nil, err
+	}
+
+	stkTx, err := s.staker.TxDetailsBTC(stkTxHash)
```

Comment:
I would say most of this logic like:
- s.staker.TxDetailsBTC

should go to `StakerApp` , in service we should only stateless parsing of arguments

Reply:
understood